packages/plugin-css/src/utils/token.ts

```diff
@@ -48,8 +48,8 @@ export function varRef(
   // suffix is only used internally (one place in plugin-sass), so handle it here rather than clutter the public API in defaultNameGenerator
   const normalizedSuffix = options?.suffix ? `-${options?.suffix.replace(DASH_PREFIX_RE, '')}` : '';
   const variableId = refID + normalizedSuffix;
-
-  return `var(${options?.generateName?.(variableId, token) ?? defaultNameGenerator(variableId, options?.prefix)})`;
+  const property = token && options?.property && typeof token.$value === 'object' && !!(token.$value as any)[options.property] ? options.property : undefined;
+  return `var(${options?.generateName?.(variableId, token) ?? defaultNameGenerator(variableId, options?.prefix)}${property ? `-${kebabinate(property)}` : ''})`;
```

Comment:
This is partially a hack, but only because of a hole in the DTCG spec. There’s currently no standard way to refer to **sub-properties** of a token, e.g. grabbing `fontFamily` from a `typography` token. Cobalt could extend the spec with this syntax, and it’s been suggested before, but I’m a little hesitant to make a decision on that because it feels imminent to happen in the spec this year.

Anyways, the “hack” is this doesn’t let people customize the final sub-property name even if they use the `generateName()` API, because there’s not an easy way to map any arbitrary string to a sub-property. Hopefully it’s not too big of a barrier for most folks.